manager/integration/tests/test_cloning.py

```diff
@@ -36,49 +94,178 @@ def test_cloning_without_backing_image():  # NOQA
+        body=clone_pvc, namespace='default')
+    wait_for_pvc_phase(core_api, clone_pvc_name, "Bound")
+
+    # Step-5
+    clone_volume_name = get_volume_name(core_api, clone_pvc_name)
+    wait_for_volume_clone_status(client, clone_volume_name, VOLUME_FIELD_STATE,
+                                 VOLUME_FIELD_CLONE_COMPLETED)
+
+    # Step-6
+    wait_for_volume_detached(client, clone_volume_name)
+
+    # Step-7,8
+    clone_pod_name = 'clone-pod' + generate_random_suffix()
+    clone_pod['metadata']['name'] = clone_pod_name
+    clone_pod['spec']['volumes'] = [create_pvc_spec(clone_pvc_name)]
+    create_and_wait_pod(core_api, clone_pod)
+    clone_data = get_pod_data_md5sum(core_api, clone_pod_name, '/data/test')
+
+    # Step-9
+    assert source_data == clone_data
+
+    # Step-10
+    wait_for_volume_healthy(client, clone_volume_name)
+
+
+@pytest.mark.cloning
+def test_cloning_with_detached_source_volume(client, core_api, pvc, clone_pvc):  # NOQA
+    """
+        1. Create a PVC:
+            ```yaml
+            apiVersion: v1
             kind: PersistentVolumeClaim
-          accessModes:
-            - ReadWriteOnce
-          resources:
-            requests:
-              storage: 10Gi
-        ```
-    14. Specify the `cloned-pvc` in a cloned pod yaml and deploy the cloned
-        pod
-    15. Wait for `source-pvc` to be attached
-    16. Wait for a new snapshot created in `source-pvc` volume created
-    17. Wait for the `CloneStatus.State` in `cloned-pvc` to be `completed`
-    18. Wait for `source-pvc` to be detached
-    19. In 3-min retry loop, wait for the cloned pod to be running
-    20. Verify the data in `cloned-pvc` is the same as in `source-pvc`
-    21. In 2-min retry loop, verify the volume of the `clone-pvc` eventually
-        becomes healthy
-    22. Cleanup the test
+            metadata:
+              name: source-pvc
+            spec:
+              storageClassName: longhorn
+              accessModes:
+                - ReadWriteOnce
+              resources:
+                requests:
+                  storage: 10Gi
+            ```
+        2. Wait for volume to be created and attach it to a node.
+        3. Write some data to the mount path of the volume
+        4. Detach the volume and wait for the volume to be in detached state.
+        5. Clone a volume by creating the PVC:
+            ```yaml
+            apiVersion: v1
+            kind: PersistentVolumeClaim
+            metadata:
+              name: cloned-pvc
+            spec:
+              storageClassName: longhorn
+              dataSource:
+                name: source-pvc
+                kind: PersistentVolumeClaim
+              accessModes:
+                - ReadWriteOnce
+              resources:
+                requests:
+                  storage: 10Gi
+            ```
+        6. Wait for `source-pvc` to be attached
+        7. Wait for a new snapshot created in `source-pvc` volume created
```

Comment:
If the cloning happens fast enough, we might not see these intermediate steps. Should we skip them?

Reply:
I ran this multiple times, we get enough time to get these states and I think these are important to check. 
If encounter problem with these checks in the Jenkins job runs, we can skip these checks.

Reply:
Okay. Thank you for the explanation